process/eng-practices/Exceptions-Haskell.md

```diff
@@ -0,0 +1,45 @@
+---
+tags:
+  - needs/review
+  - needs/writing
+---
+# Exceptions in Haskell
+
+In `cardano-wallet`, we currently use the following conventions for dealing with exceptions:
+
+1. Avoid partial functions.\
+    Example: Do not use `head`, but use a pattern match, or consider using functions from `Safe` module, such as `headMay`. Considering all corner cases does catch bugs.
```

Comment:
Here's another example that is a little unexpected: `Data.Text.Encoding.decodeUtf8`. It crashes with invalid input.